src/server/zset_family.cc

```diff
@@ -2030,6 +2029,59 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInter(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<SetOpArgs> op_args_res = ParseSetOpArgs(args, false);
+
+  if (!op_args_res) {
+    return HandleOpStatus(cntx, op_args_res.status());
+  }
+  const auto& op_args = *op_args_res;
+  if (op_args.num_keys == 0) {
+    return SendAtLeastOneKeyError(cntx);
+  }
+
+  vector<OpResult<ScoredMap>> maps(shard_set->size(), OpStatus::SKIPPED);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpInter(shard, t, "", op_args.agg_type, op_args.weights, false);
+    return maps[shard->shard_id()];
+  };
+
+  cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  ScoredMap result;
+  for (auto& op_res : maps) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res)
+      return (*cntx)->SendError(op_res.status());
+
+    if (op_res->empty()) {
+      result.clear();
+      break;
+    }
+
+    if (result.empty()) {
+      result.swap(op_res.value());
+    } else {
+      InterScoredMap(&result, &op_res.value(), op_args.agg_type);
+    }
+  }
```

Comment:
actually, this loop also appears multiple times.
we could introduce `OpResult<ScoredMap> IntersectResults(const vector<OpResult<ScoredMap>>& results)` function for that.

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/fc5db570b8aefb3980ba6dffd33a8139ebf37e0e